src/azure-cli/azure/cli/command_modules/ams/tests/latest/test_ams_sp_scenarios.py

```diff
@@ -14,7 +14,7 @@ class AmsSpTests(ScenarioTest):
     @StorageAccountPreparer(parameter_name='storage_account_for_create')
     @AllowLargeResponse()
     def test_ams_sp_create_reset(self, resource_group, storage_account_for_create):
-        with mock.patch('azure.cli.command_modules.ams._utils._gen_guid', side_effect=self.create_guid):
+        with mock.patch('azure.cli.command_modules.ams.operations.sp._gen_guid', side_effect=self.create_guid):
```

Comment:
Follow [Where to patch](https://docs.python.org/3/library/unittest.mock.html#where-to-patch)